### DISCUSSION TITLE
How to run `pub upgrade` for all `pubspec.yaml` files in Serverpod

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#1884] [General]
Currently the `util/generate_all` script runs `pub get` for each project. However, I need to run `pub upgrade` to fix some version issues.

I tried this command:

```bash
find . -name 'pubspec.yaml' -execdir dart pub upgrade \;
```

but a lot of the `pubspec.yaml` files apparently need preprocessing, e.g.:

```console
Resolving dependencies... 
Error on line 5, column 10 of pubspec.yaml: Invalid version number: Could not parse "VERSION".
  ╷
5 │ version: VERSION
  │          ^^^^^^^
  ╵
Error on line 4, column 5 of pubspec.yaml: Mapping values are not allowed here. Did you miss a colon earlier?
  ╷
4 │ name: serverpod_auth_apple_flutter
  │     ^
  ╵
```

**Comment by Isakdl** [comment]
The script in `util/pub_get_all` will run a `dart pub upgrade` for all packages.

And you have `util/update_pubspecs` if you change the pubspec templates.

**Reply by lukehutch** [reply]
Ah, I need to update my knowledge of these scripts! Thanks.

What exactly are these templates, if they don't start out as valid `pubspec.yaml` files? Is this a Serverpod thing?

**Reply by vlidholt** [reply]
We used the templates more extensively previously, but moved over to pubspec overrides recently. Right now, I think it's only used to modify the version of the templates. We are looking into finding a better solution for this. And yes, it's a Serverpod thing.

**Reply by lukehutch** [reply]
@vlidholt try just using regular yaml, and then overwriting the version code when necessary using `yq`:

https://github.com/mikefarah/yq

Orthogonally related: I used `yq` to create a database login script:

```bash
export PGPASSWORD="$(yq -r .development.database config/passwords.yaml)"
psql -h localhost -p 8090 -U postgres --dbname click
```
